### PR TITLE
boards/nucleo144-f767: add initial support

### PR DESCRIPTION
**Pull request by aabadie**
I based this PR on #6907 since it introduces support for stm32f7. The UART configuration is the same and this makes me think that we could factorize them.

By the way, similar to the problem I raised in #6907, I don't have UART or can't make a led blink.

### CODE REVIEW

**Comment by aabadie** [comment]
Rebased. Just tested it and works. I had to install gcc-arm_embedded from launchpad (now I have gcc 6.3.1).

**Comment by astralien3000** [comment]
I tested : hello-world, default (saul leds+button), periph_timer, periph_uart, xtimer_usleep.
All is ok.

**Comment by aabadie** [comment]
Thanks @astralien3000 !

Just a review is missing now. @haukepetersen, @vincent-d  or @astralien3000 ?

File: cpu/stm32f7/include/cpu_conf.h [hunk 1]
```diff
@@ -40,7 +42,7 @@ extern "C" {
 #define CPU_DEFAULT_IRQ_PRIO            (1U)
 #if defined(CPU_MODEL_STM32F746ZG)
 #define CPU_IRQ_NUMOF                   (98U)
-#elif defined(CPU_MODEL_STM32F769NI)
+#elif defined(CPU_MODEL_STM32F769NI) || defined(CPU_MODEL_STM32F767ZI)
 #define CPU_IRQ_NUMOF                   (110U)
```
Comment (vincent-d) [inline cpu/stm32f7/include/cpu_conf.h]: Vectors definition needs update

Comment (aabadie) [inline cpu/stm32f7/include/cpu_conf.h]: done

Comment (vincent-d) [inline cpu/stm32f7/include/cpu_conf.h]: Function declarations are missing ;)

Comment (aabadie) [inline cpu/stm32f7/include/cpu_conf.h]: oups! fixed

**Comment by aabadie** [comment]
squashed

**Comment by aabadie** [comment]
all green @vincent-d !

**Comment by vincent-d** [comment]
Go!

**Pull request merged**